tests/e2e/operator/operator_test.go

```diff
@@ -99,33 +107,58 @@ var _ = Describe("Operator", Ordered, func() {
+				}
+
+				g.Expect(crdNames).To(ContainElements(crds), "Istio CRDs are not present; expected list to contain all elements")
+			}).Should(Succeed(), "Unexpected error getting CRDs from the cluster")
 			Success("Istio CRDs are present")
 		})
 
 		It("updates the CRDs status to Established", func() {
 			for _, crd := range crds {
-				Eventually(kubectl.GetConditions).WithArguments(namespace, "crd", crd).Should(ContainElement(crdEstablished), "CRD is not Established")
+				Eventually(func(g Gomega) {
+					Expect(k8sClient.Get(context.TODO(), client.ObjectKey{Namespace: namespace, Name: crd}, customResourceDefintion)).
+						To(Succeed(), "Error getting CRD")
+
+					for _, condition := range customResourceDefintion.Status.Conditions {
+						if condition.Type == "Established" {
+							g.Expect(string(condition.Status)).To(Equal("True"), "CRD conditions was not the expected")
+						}
+					}
+				}).Should(Succeed(), "CRD is not Established")
 			}
 			Success("CRDs are Established")
 		})
 
 		Specify("istio crd is present", func() {
 			// When the operator runs in OCP cluster, the CRD is created but not available at the moment
-			Eventually(kubectl.GetResourceList).
-				WithArguments("", "istio").
-				Should(Equal(kubectl.EmptyResourceList), "Istio CRD is not present; expected to not fail and return an empty list of Istio CR")
+			Eventually(k8sClient.Get).WithArguments(context.TODO(), client.ObjectKey{Name: "istios.operator.istio.io"}, customResourceDefintion).
+				Should(Succeed(), "Error getting Istio CRD")
 			Success("Istio CRD is present")
 		})
 
 		It("starts successfully", func() {
-			Eventually(kubectl.GetConditions).
-				WithArguments(namespace, "deployment", deploymentName).
-				Should(ContainElement(resourceAvailable), "Operator deployment is not Available; unexpected Condition")
-			Success("Operator deployment is Available")
-
-			Expect(kubectl.GetPodPhase(namespace, "control-plane=sail-operator")).To(Equal("Running"), "Operator failed to start; unexpected pod Phase")
+			Eventually(func(g Gomega) {
+				g.Expect(k8sClient.Get(context.TODO(), client.ObjectKey{Namespace: namespace, Name: deploymentName}, deployment)).
+					To(Succeed(), "Error getting operator deployment")
+				for _, condition := range deployment.Status.Conditions {
+					if condition.Type == "Available" {
+						g.Expect(string(condition.Status)).To(Equal("True"), "Operator conditions was not the expected")
+					}
+				}
+			}).Should(Succeed(), "Operator deployment is not Available")
+
+			selector := client.MatchingLabels(map[string]string{"control-plane": "sail-operator"})
+			Expect(k8sClient.List(context.TODO(), podList, selector, client.InNamespace(namespace))).To(Succeed(), "Error getting operator pod")
+
+			Expect(podList.Items).To(HaveLen(1), "Unexpected number of operator pods; expected list to have length 1")
+			Expect(string(podList.Items[0].Status.Phase)).To(Equal("Running"), "Operator failed to start; unexpected pod Phase")
```

Comment:
Since we're already checking the deployment's Available condition, we don't need to check the pod phase.